cpp/include/rapidsmpf/buffer/packed_data.hpp

```diff
@@ -74,9 +76,16 @@ struct PackedData {
      * @brief Check if the packed data is empty.
      *
      * @return True if the packed data is empty, false otherwise.
+     *
+     * @throw std::invalid_argument if metadata and gpu_data is null and non-null.
      */
     [[nodiscard]] bool empty() const {
-        return metadata->empty() && gpu_data->size() == 0;
+        RAPIDSMPF_EXPECTS(
+            (!metadata) == (!gpu_data),
+            "the metadata and gpu_data pointers cannot be null and non-null",
+            std::invalid_argument
+        );
+        return metadata->empty() && gpu_data->size == 0;
```

Comment:
*question*: Can a packed data ever have both `metadata` and `gpu_data` as nullptr?